java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java

```diff
@@ -441,7 +534,30 @@ public File writeObf(Map<String, GpxFile> gpxFiles, List<KFile> files, File tmpF
 		}
 		return targetObf;
 	}
-	
+
+	private void generateUniqueBaseOsmId(Map<String, GpxFile> gpxFiles, List<KFile> kFiles) {
+		int seed = 0;
+		if (gpxFiles != null) {
+			seed += gpxFiles.size();
+			for (String name : gpxFiles.keySet()) {
+				for (int i = 0; i < name.length(); i++) {
+					seed += name.charAt(i);
+				}
+			}
+		}
+		if (kFiles != null) {
+			for(KFile f : kFiles) {
+				String name = f.name();
+				for (int i = 0; i < name.length(); i++) {
+					seed += name.charAt(i);
+				}
+			}
+		}
+		if (seed != 0) {
+			baseOsmId = -(new Random(seed).nextLong() & baseIdMask);
```

Comment:
Need to discuss about id for Travel OBF gpx

Reply:
random baseOsmId is removed (will be a problem for combined gpx collections)